lib/web_plugin_internal/FfJavascriptSDKInterop.dart

```diff
@@ -0,0 +1,44 @@
+@JS()
+library harness_javascript_sdk.js;
+
+import 'package:js/js.dart';
+
+// Represents the JavaScript SDK global object set by the iife distribution.
+@JS('HarnessFFSDK')
+class JavaScriptSDK {
+  static const windowReference = 'HarnessFFSDK';
+  static const initializeFunction = 'initialize';
+
+  external static dynamic initialize(
+      String apiKey, Map<String, dynamic> target, Map<String, dynamic> options);
+}
+
+// Represents the JavaScript SDK Client instance. Once we've initialized the
+// Client, we set it as a property on the window using this reference.
+@JS('cfClient')
+class JavaScriptSDKClient {
+  static const windowReference = 'cfClient';
+  static const onFunction = 'on';
+  static const offFunction = 'off';
+  static const variation = 'variation';
+  static const close = 'close';
+
+  external static dynamic on(dynamic eventType, Function callback);
+  external static dynamic off(dynamic eventType, Function callback);
+}
+
+// Represents the events that the JavaScript SDK Client can emit
+class Event {
```

Comment:
Enums are a bit clunky to use as Strings in dart. They do have a to.String() method, but that returns the name and value, so it has to be split. I preferred a class of static consts for the data type.